### PR TITLE
contributing: Improve where doc is located

### PR DESCRIPTION
**Pull request by wenzeslaus**
* Move C API doc compilation from readme.
* Move compilation intro to the end of contributing file.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -106,6 +81,10 @@ git fetch upstream
 git rebase upstream/main
 ```
 
+Notably, you should not make commits to your main branch,
```
Comment (veroandreo) [inline CONTRIBUTING.md:84]:
```suggestion
Notably, you should not make commits to your local main branch,
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -106,6 +81,10 @@ git fetch upstream
...
+so the above is than just a simple update (and no actual
```
Comment (veroandreo) [inline CONTRIBUTING.md:85]:
```suggestion
so the above is then just a simple update (and no actual
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -106,6 +81,10 @@ git fetch upstream
...
+rebase or merge happens).
+
 ### Update if you have local branches
```
Comment (veroandreo) [inline CONTRIBUTING.md:88]:
```suggestion
### Update if you have local changes
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -151,6 +124,27 @@ in the local copy of the code. When make changes, please follow
...
 
+### Testing changes
+
+Testing helps to ensure that the changes work well with the rest
+of the project. While there is many different ways to test,
+usually you will want to compile the source code (see below),
+add test code (using _grass.gunittest_ or pytest), and run code
+linters (automated code quality checks).
+
+There is a series of automated checks which will run you pull request
```
Comment (veroandreo) [inline CONTRIBUTING.md:135]:
```suggestion
There is a series of automated checks which will run on your pull request
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -151,6 +124,27 @@ in the local copy of the code. When make changes, please follow
...
+after you create that later on. You don't need to run every one of these
```
Comment (veroandreo) [inline CONTRIBUTING.md:136]:
```suggestion
after you create one. You don't need to run all these
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -151,6 +124,27 @@ in the local copy of the code. When make changes, please follow
...
+checks locally and some of these may fail for your code. This is a part of
```
Comment (veroandreo) [inline CONTRIBUTING.md:137]:
```suggestion
checks locally and, indeed, some of them may fail for your code. This is a part of
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -151,6 +124,27 @@ in the local copy of the code. When make changes, please follow
...
+the standard iterative process of interating changes into the main code,
```
Comment (veroandreo) [inline CONTRIBUTING.md:138]:
```suggestion
the standard iterative process of integrating changes into the main code,
```

**Review by veroandreo — CHANGES_REQUESTED** [review]
I'll continue after the Ice-breaker.... ehehehe

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -151,6 +124,27 @@ in the local copy of the code. When make changes, please follow
...
+usually you will want to compile the source code (see below),
+add test code (using _grass.gunittest_ or pytest), and run code
+linters (automated code quality checks).
+
+There is a series of automated checks which will run on your pull request
+after you create one. You don't need to run all these
+checks locally and, indeed, some of them may fail for your code. This is a part of
+the standard iterative process of integrating changes into the main code,
+so if that happens, just see the error messages and go back to your code
+and try again. If you are not sure what to do, let other know in a pull
```
Comment (veroandreo) [inline CONTRIBUTING.md:140]:
```suggestion
so if that happens, just see the error messages, go back to your code
and try again. If you are not sure what to do, let others know in a pull
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -151,6 +124,27 @@ in the local copy of the code. When make changes, please follow
...
+request comment.
+
+There are some steps you can do locally, to improve your code.
```
Comment (veroandreo) [inline CONTRIBUTING.md:143]:
```suggestion
Note that there are some steps you can do locally to improve your code.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -151,6 +124,27 @@ in the local copy of the code. When make changes, please follow
...
+For Python, run `black .` to apply standardized formatting. You can
+also run linter tools such as Pylint which will suggest you how
+you may improve your code.
```
Comment (veroandreo) [inline CONTRIBUTING.md:146]:
```suggestion
also run linter tools such as Pylint which will suggest different improvements 
to your code.
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -198,3 +195,29 @@ local _main_ branch in order to get the change you just contributed.
...
+a runnable binary program from the source code,
+using the so-called "compilation step". While the
+source code consists of thousands of C and Python files (plus HTML
+documentation and other files), the included "makefiles" tell the build system to
+generate binaries from the source code in the correct order, render the
+manual pages, etc.
+
+The way to install the compiler tools and Python depends on the operating
+system. To make this easier, we have collected copy-paste instructions
+for most operating systems in our wiki:
```
Comment (veroandreo) [inline CONTRIBUTING.md:211]:
```suggestion
to install dependencies and compile GRASS source code for most operating systems.
Please see our dedicated wiki:
```

**Review by veroandreo — COMMENTED** [review]
done :)

**Review by veroandreo — APPROVED** [review]

**Review by neteler — APPROVED** [review]

**Pull request merged**